src/encoding.rs

```diff
@@ -934,26 +924,20 @@ impl sealed::BytesAdapter for Vec<u8> {
         self.reserve(buf.remaining());
         self.put(buf);
     }
-
-    fn append_to<B>(&self, buf: &mut B)
-    where
-        B: BufMut,
-    {
-        buf.put(self.as_slice())
-    }
 }
 
 pub mod bytes {
     use super::*;
 
     pub fn encode<A, B>(tag: u32, value: &A, buf: &mut B)
     where
-        A: BytesAdapter,
+        A: AsRef<[u8]>,
```

Comment:
I would like to see tests to guarantee `impl BytesAdapter`, `Bytes` and `Vec<u8>` can still be used on this function.

And you need to add tests for your new use case, to make sure it is not broken in the future.